lib/apiservers/portlayer/restapi/handlers/containers_handlers.go

```diff
@@ -360,7 +361,10 @@ func (handler *ContainersHandlersImpl) GetContainerLogsHandler(params containers
 		return containers.NewGetContainerLogsInternalServerError()
 	}
 
-	detachableOut := NewFlushingReader(reader)
+	// wrap the reader in a LogReader to pull log messages out of JSON entries
+	lr := ioutil.NewLogReader(reader)
```

Comment:
As the container log format is changed, portlayer will have to be backward compatible, otherwise the container log does not work for old version's container.
So will suggest you add interface for the LogReader and LogWriter. For old version's container, text file reader should still be there.
And also you'll need one upgrade case to make sure the function for old container still works